File: python/Product.py
class Product:

    def __init__(self, id = "", name = "", brand = "", price = 0):
        self.__id = id
        self.__name = name
        self.__brand = brand
        self.__price = price

    def getId(self):
        return self.__id
    
    def getName(self):
        return self.__name
    
    def getBrand(self):
        return self.__brand
    
    def getPrice(self):
        return self.__price
    
    def setId(self, id):
        self.__id = id

    def setName(self, name):
        self.__name = name

    def setBrand(self, brand):
        self.__brand = brand

    def setPrice(self, price):
        self.__price = price


    

    

    

File: cpp/Product.cpp
#include <iostream>
#include <string>

class Product {
    private:
        std::string id;
        std::string name;
        std::string brand;
        int price;

    public:
        // Constructor
        Product(const std::string& id = "", const std::string& name = "", const std::string& brand = "", int price = 0.0) 
            : id(id), name(name), brand(brand), price(price) {}

        // Getter methods
        std::string getId() const { return id; }
        std::string getName() const { return name; }
        std::string getBrand() const { return brand; }
        int getPrice() const { return price; }

        // Setter methods
        void setId(const std::string& newId) { id = newId; }
        void setName(const std::string& newName) { name = newName; }
        void setBrand(const std::string& newBrand) { brand = newBrand; }
        void setPrice(int newPrice) { price = newPrice; }
};
